### ISSUE TITLE
Make metrics filter for generic scraper optional

### ISSUE THREAD

**Issue by jdevalk2** [tomkerkhove/promitor#192]
## Expected Behavior
When defining a new generic resource to scrape..

I am expecting to be able to leave the filter empty, or null.
Deep down, where it constructs the query to Azure Monitor the filter will then not be set resulting in a successful query. It actually checks to see if the property is null. The validation just won't allow it.

For instance, querying the network bytes on a Postgres database
```
metrics: 
- name: "postgres_server_network_in"
  description: "Postgres network in"
  resourceType: "Generic"
  resourceUri: "Microsoft.DBforPostgreSQL/servers/mypostgresdbname"
  filter: "1 eq 1"
  azureMetricConfiguration:
    metricName: "network_bytes_ingress"
    aggregation: "None"
```

## Actual Behavior
The validation fails if the filter parameter is not set.

## Steps to Reproduce the Problem

  1. Create postgres DB in azure
  2. Configure with snippet above
  3. Exception is thrown when trying to retrieve the metric

## Specifications

  - Version: 0.2.1 (latest)
  - Kubernetes AKS on Azure test cluster

I want to be able to give it this, which I think will fix the problem:

```
metrics: 
- name: "postgres_server_network_in"
  description: "Postgres network in"
  resourceType: "Generic"
  resourceUri: "Microsoft.DBforPostgreSQL/servers/mypostgresdbname"
  azureMetricConfiguration:
    metricName: "network_bytes_ingress"
    aggregation: "None"
```

**Comment by tomkerkhove** [comment]
This should be fixed in #194.

The following sample works:
```yaml
azureMetadata:
  tenantId: c8819874-9e56-4e3f-b1a8-1c0325138f27
  subscriptionId: 0f9d7fea-99e8-4768-8672-06a28514f77e
  resourceGroupName: promitor
metrics:
  - name: demo_generic_namespace_size
    description: "Size of all queues in our Azure Service Bus namespace (determined with Generic provider)"
    resourceType: Generic
    resourceUri: Microsoft.ServiceBus/namespaces/promitor-messaging
    azureMetricConfiguration:
      metricName: ActiveMessages
      aggregation: Total
```

**Comment by jdevalk2** [comment]
I will give the new version a try. Thanks for the update.

**Comment by tomkerkhove** [comment]
Feel free to re-open if it is not fixed for you @jdevalk2

**Comment by jungopro** [comment]
hi @tomkerkhove I'm using the docker image v0.2.1, did you implement that fix in the docker image?
If not, where can I find a dockerfile to build it?

Thanks
Omer

**Comment by tomkerkhove** [comment]
hi @jungopro, this is not yet released but you can use the Dockerfile in https://github.com/tomkerkhove/promitor/blob/master/src/Promitor.Scraper.Host/Dockerfile.

I'll release a new version later today.

**Comment by tomkerkhove** [comment]
This is available in v0.3.0 - https://github.com/tomkerkhove/promitor/releases/tag/0.3.0
